jOOL-java-8/src/main/java/org/jooq/lambda/Agg.java

```diff
@@ -906,11 +906,27 @@ else if (l1[0] == null)
 
     /**
      * Get a {@link Collector} that calculates the derived <code>PERCENTILE_DISC(percentile)</code> function given a specific ordering.
+     *
+     * @param function map the items in the streams into values
+     * @param comparator comparator used for sorting the items
+     * @return a collector that calculates the derived <code>PERCENTILE_DISC(percentile)</code> function
      */
     public static <T, U> Collector<T, ?, Optional<T>> percentileBy(double percentile, Function<? super T, ? extends U> function, Comparator<? super U> comparator) {
         if (percentile < 0.0 || percentile > 1.0)
             throw new IllegalArgumentException("Percentile must be between 0.0 and 1.0");
 
+        // CS304 Issue link: https://github.com/jOOQ/jOOL/issues/376
+        if (percentile == 0.0)
+            // If percentile is 0, this is the same as taking the item with the minimum value.
+            return minBy(function, comparator);
```

Comment:
The comment says the same thing as the method call, so it isn't really necessary. I'll remove it again after merging.